pkg/controller/service/util/util.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
```

Comment:
One more nit (more of a personal preference): I don't think a `util` package is necessary for patch, can just have a `patch.go` go file in `pkg/controller/service/`

Reply:
Agree. Fewer public methods to accumulate callers, the better

Reply:
Thanks, good point. Put it into `patch.go` instead.